quinn-proto/src/congestion/bbr/mod.rs

```diff
@@ -468,6 +469,13 @@ impl Controller for Bbr {
         self.loss_state.lost_bytes += lost_bytes;
     }
 
+    fn on_mtu_update(&mut self, current_mtu: u16) {
+        self.current_mtu = current_mtu as u64;
+        self.min_cwnd = calculate_min_window(self.current_mtu);
+        self.init_cwnd = self.init_cwnd.max(self.min_cwnd);
```

Comment:
This means `init_cwnd` can only ever increase, not shrink, even if the MTU decreases. Is that intended? We don't seem to be updating initial window at all in the other algorithms.

Reply:
Hmmm... There seems to be a difference in the way the initial window is used in bbr compared to cubic and new reno. The latter two only use it upon creating the controller, so there is no need to update it, but bbr is using it in more places, so it makes sense to update it.

You are right that the mentioned code will only increase `init_cwnd` and never shrink it, which is a mistake. We actually want to clamp `config.initial_window` between `min_cwnd` and `u64::max` (I just pushed a commit doing so)